testing/test_cacheprovider.py

```diff
@@ -44,52 +44,54 @@ def test_cache_writefail_cachfile_silent(self, pytester: Pytester) -> None:
         assert cache is not None
         cache.set("test/broken", [])
 
-    @pytest.mark.skipif(sys.platform.startswith("win"), reason="no chmod on windows")
-    @pytest.mark.filterwarnings(
-        "ignore:could not create cache path:pytest.PytestWarning"
-    )
-    def test_cache_writefail_permissions(self, pytester: Pytester) -> None:
-        pytester.makeini("[pytest]")
+    @pytest.fixture
+    def unwritable_cache_dir(self, pytester: Pytester) -> Generator[Path, None, None]:
         cache_dir = pytester.path.joinpath(".pytest_cache")
         cache_dir.mkdir()
         mode = cache_dir.stat().st_mode
         cache_dir.chmod(0)
-        try:
-            config = pytester.parseconfigure()
-            cache = config.cache
-            assert cache is not None
-            cache.set("test/broken", [])
-        finally:
-            cache_dir.chmod(mode)
+        if os.access(cache_dir, os.W_OK):
+            pytest.skip("Failed to make cache dir unwritable")
```

Comment:
maybe xfail, but ok as skip as well

Reply:
I don't think xfail is accurate. The test didn't fail, it can't run at all because of environmental stuff.

Reply:
personallyi lean on xfail due to being mid test and noting "yup, i can't possibly pass this, i got the wrong thing to work with"

Reply:
I still don't see how it's mid-test - quite the opposite, it's in a fixture, and it's a prerequisite for the test which failed. I'd agree if we wanted to test if pytest somehow chmod's one of it's directories, but here we just need the directory to test something else. It's essentially the same like the skip before, where we just hardcoded Windows rather than doing a feature check.

Given that you said skip is okay for you as well and nobody else complained, I'm taking the freedom to merge this as-is.